File: frontend/src/types/notification.ts
export type NotificationType =
  | 'LEAVE_REQUEST'
  | 'APPROVAL'
  | 'ANNOUNCEMENT'
  | 'PAYROLL'
  | 'ATTENDANCE'
  | 'TIME_ADJUSTMENT'
  | 'GENERAL';

export interface NotificationItem {
  id: number;
  userId: number;
  notificationType: string;
  title: string;
  message?: string | null;
  referenceType?: string | null;
  referenceId?: number | null;
  isRead: boolean;
  createdAt: string;
  readAt?: string | null;
  targetUrl: string;
  timeAgo: string;
}

export interface NotificationSummary {
  unreadCount: number;
  recentNotifications: NotificationItem[];
}

File: backend/src/Hrms.Domain/Entities/Notification.cs
using Hrms.Domain.Common;

namespace Hrms.Domain.Entities;

/// <summary>
/// ข้อมูลการแจ้งเตือนภายในระบบ (In-App Notification)
/// แมปกับตาราง hrms.notification
/// </summary>
public class Notification : BaseEntity
{
    public long UserId { get; set; }
    public string NotificationType { get; set; } = string.Empty; // LEAVE_REQUEST, APPROVAL, ANNOUNCEMENT, PAYROLL, ATTENDANCE, GENERAL
    public string Title { get; set; } = string.Empty;
    public string? Message { get; set; }
    public string? ReferenceType { get; set; } // LEAVE_REQUEST, ANNOUNCEMENT, PAYROLL_PERIOD, ATTENDANCE
    public long? ReferenceId { get; set; }
    public bool IsRead { get; set; } = false;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? ReadAt { get; set; }

    // Navigation Property
    public virtual UserAccount User { get; set; } = null!;
}
